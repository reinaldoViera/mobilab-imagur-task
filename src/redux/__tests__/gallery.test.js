import reducer, {
    constants
} from '../gallery';


describe('Gallery reducer', () => {
    it('should give initial state', () => {
        let state = reducer(undefined, {
            type: "FAKE"
        });
        expect(state).toMatchSnapshot();
    });
    it('should add galleries to existing', () => {
        let state = reducer({
            galleries: {
                "khbckdjw": {
                    fake: 'data'
                },
                "jehfindcs": {
                    more: 'fake'
                }
            },
            list: ["jehfindcs", "khbckdjw"],
            nextPage: 1
        }, {
            type: constants.ADD_GALLERIES,
            payload: {
                galleries: {
                    "viwuywi": {
                        new_fake: 'data'
                    },
                    "gefgegr": {
                        new_more: 'fake'
                    }
                },
                list: ["jehfindcs", "khbckdjw"]
            }
        });
        expect(state).toMatchSnapshot();
    });
    it('should replace galleries to existing', () => {
        let state = reducer({
            galleries: {
                "khbckdjw": {
                    fake: 'data'
                },
                "jehfindcs": {
                    more: 'fake'
                }
            },
            list: ["jehfindcs", "khbckdjw"],
            nextPage: 1
        }, {
            type: constants.ADD_GALLERIES,
            payload: {
                galleries: {
                    "viwuywi": {
                        new_fake: 'data'
                    },
                    "gefgegr": {
                        new_more: 'fake'
                    }
                },
                list: ["jehfindcs", "khbckdjw"],
                replace: true
            }
        });
        expect(state).toMatchSnapshot();
    });

    it('should avoid "rising" sort if is not section "user"', () => {
        let state = reducer({
            section: 'user',
            viral: false,
            window: 'day',
            sort: 'rising',
            loading: false,
            error: false,
            nextPage: 0,
            list: [],
            galleries: {}
          }, {
            type: constants.CHANGE_SECTION,
            payload: {
                section: 'top',
                sort: 'rising'
            }
        });
        expect(state).toMatchSnapshot();
    });

})