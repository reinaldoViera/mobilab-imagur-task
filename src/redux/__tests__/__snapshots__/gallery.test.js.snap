// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery reducer should add galleries to existing 1`] = `
Object {
  "galleries": Object {
    "gefgegr": Object {
      "new_more": "fake",
    },
    "jehfindcs": Object {
      "more": "fake",
    },
    "khbckdjw": Object {
      "fake": "data",
    },
    "viwuywi": Object {
      "new_fake": "data",
    },
  },
  "list": Array [
    "jehfindcs",
    "khbckdjw",
    "jehfindcs",
    "khbckdjw",
  ],
  "loading": false,
  "nextPage": 2,
}
`;

exports[`Gallery reducer should avoid "rising" sort if is not section "user" 1`] = `
Object {
  "error": false,
  "galleries": Object {},
  "list": Array [],
  "loading": false,
  "nextPage": 0,
  "section": "top",
  "sort": "viral",
  "viral": false,
  "window": "day",
}
`;

exports[`Gallery reducer should give initial state 1`] = `
Object {
  "error": false,
  "galleries": Object {},
  "list": Array [],
  "loading": false,
  "nextPage": 0,
  "section": "hot",
  "sort": "viral",
  "viral": true,
  "window": "day",
}
`;

exports[`Gallery reducer should replace galleries to existing 1`] = `
Object {
  "galleries": Object {
    "gefgegr": Object {
      "new_more": "fake",
    },
    "viwuywi": Object {
      "new_fake": "data",
    },
  },
  "list": Array [
    "jehfindcs",
    "khbckdjw",
  ],
  "loading": false,
  "nextPage": 0,
}
`;
